vault/resource_terraform_cloud_secret_creds.go

```diff
@@ -0,0 +1,109 @@
+package vault
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/vault/api"
+)
+
+func terraformCloudSecretCredsResource() *schema.Resource {
+	return &schema.Resource{
+		Create: readTerraformCloudSecretCredsResource,
+		Read:   readTerraformCloudSecretCredsResource,
```

Comment:
I don't think `Create` and `Read` can be the same function here. For `Read` we need to read the token's lease ID to see if the token is still valid and exists. If not, then we will remove it from state with `d.SetId("")`. Alternatively we could do this by implementing an `Exists` function. 

As this stands, every `plan` (which does `Read`) is generating a token, and subsequent (if separate) `apply` also generates a token.